packages/editor/src/components/parts/program/editor/EditorPart.tsx

```diff
@@ -0,0 +1,77 @@
+import { EditorData, Label, Script, Text, Widget } from '@axonivy/inscription-protocol';
+import { PartProps, usePartDirty, usePartState } from '../../../editors';
+import { useEditorContext, useMeta, useValidations } from '../../../../context';
+import { useEditorData } from './useEditorData';
+import { Input, MessageText, ScriptInput } from '../../../../components/widgets';
+
+export function useEditorPart(): PartProps {
+  const { config, defaultConfig, initConfig, reset } = useEditorData();
+  const compareData = (data: EditorData) => [data.userConfig, data.link];
+  const validation = useValidations(['javaClass']);
+  const state = usePartState(compareData(defaultConfig), compareData(config), validation);
+  const dirty = usePartDirty(compareData(initConfig), compareData(config));
+  return {
+    name: 'Editor',
+    state,
+    reset: { dirty, action: () => reset() },
+    content: <EditorPart />
+  };
+}
+
+const EditorPart = () => {
+  const { config, update } = useEditorData();
+  const { context } = useEditorContext();
+  const editorItems = useMeta('meta/program/editor', { context, type: config.javaClass }, []);

```

Comment:
directly get the data here instead of everywhere below:
`const editorItems = useMeta('meta/program/editor', { context, type: config.javaClass }, []).data;`